### PR TITLE
parallelize the flush

### PR DESCRIPTION
**Pull request by tobowers**
This is a simple fix that paralellizes the flush
Before:
```
$ go test . -run ^TestSetGet -v -count=1
=== RUN   TestSetGet
Total size is: 7740530, size of keys+vals: 6600000, overhead: 1.17
&{4713 100000 map[1:41261 2:16450 3:8613]}
start flush
flush took:  594.296877ms
finds took:  815.670497ms
--- PASS: TestSetGet (3.93s)
PASS
ok  	github.com/quorumcontrol/go-hamt-ipld	3.949s
```

After:
```
go test . -run ^TestSetGet -v -count=1
=== RUN   TestSetGet
Total size is: 7740530, size of keys+vals: 6600000, overhead: 1.17
&{4713 100000 map[1:41261 2:16450 3:8613]}
start flush
flush took:  213.139625ms
finds took:  904.898586ms
--- PASS: TestSetGet (3.76s)
PASS
ok  	github.com/quorumcontrol/go-hamt-ipld	3.791s
```

### CODE REVIEW

File: hamt.go [hunk 1]
```diff
@@ -150,21 +151,27 @@ func (n *Node) checkSize(ctx context.Context) (uint64, error) {
...
 
 func (n *Node) Flush(ctx context.Context) error {
+	var wg sync.WaitGroup
 	for _, p := range n.Pointers {
 		if p.cache != nil {
-			if err := p.cache.Flush(ctx); err != nil {
-				return err
-			}
+			wg.Add(1)
+			go func(p *Pointer) {
```
Comment (cap10morgan) [inline hamt.go:160]: A Pointer pointer! 👉 👈

File: hamt.go [hunk 2]
```diff
@@ -150,21 +151,27 @@ func (n *Node) checkSize(ctx context.Context) (uint64, error) {
...
+				if err := p.cache.Flush(ctx); err != nil {
+					// return err
```
Comment (cap10morgan) [inline hamt.go]: Do we need to do something with these errors now?

**Review by cap10morgan — COMMENTED** [review]
Question about error handling in the go func.

File: hamt.go (hunk 2 above)
Comment (tobowers) [reply hamt.go]: oh yes we do

Comment (tobowers) [reply hamt.go]: and fixorated

**Review by cap10morgan — APPROVED** [review]
⏩

**Pull request merged**